projects/wave-core-new/src/lib/backend/backend.service.ts

```diff
@@ -241,6 +244,20 @@ export class BackendService {
         });
     }
 
+    upload(sessionId: UUID, form: FormData): Observable<HttpEvent<UploadResponseDict>> {
+        return this.http.post<UploadResponseDict>(this.config.API_URL + '/upload', form, {
+            headers: BackendService.authorizationHeader(sessionId),
+            reportProgress: true,
+            observe: 'events',
+        });
+    }
+
+    createDataSet(sessionId: UUID, createDataSet: CreateDataSetDict): Observable<DataSetIdDict> {
```

Comment:
can't we go over to `createDataset`? :wink:

Reply:
yeah I will do a PR at some point unifying it across front and backend to Dataset